### DISCUSSION TITLE
ol and ol-mapbox-style package

### DISCUSSION THREAD

**Discussion by Wockeez** [openlayers/openlayers#14973] [Q&A] [answered]
I was looking at the package.json of these two packages and noticed:

- ol package depends on ol-mapbox-style
- ol-mapbox-style depends on the ol package

**It's a cyclic dependency.**

This may be safe up to one point

Suppose I forked these packages and prefixed them with `@lib/ol` `@lib/ol-mapbox-style` and assigned the registry `@lib/registry` (and this registry is empty).

Since one library depends on the other, it will be impossible to build either of the two packages because the code depends on each other. It's a hopeless situation

Question: ol-mapbox-style was written to isolate the code (this is true, but not when it depends on the source) why was it necessary to write an ol-mapbox-style library and not integrate it into the ol package?

**Comment by ahocevar** [comment, marked as answer]
This will be fixed in the next release with https://github.com/openlayers/openlayers/pull/14870.

To answer your question: ol-mapbox-style was created as a 3rd-party library to extend OpenLayers's functionality. Later, `ol/layer/MapboxVector` was created in OpenLayers for convenience. The mentioned pull request removes `ol/layer/MapboxVector`, because ol-mapbox-style now provides an equally convenient way to do the same.

**Reply by Wockeez** [reply]
Thanks
